dcpy/lifecycle/package/validate.py

```diff
@@ -0,0 +1,302 @@
+from dataclasses import dataclass, field
+from enum import Enum
+import geopandas as gpd
+import pandas as pd
+from pathlib import Path
+import pprint
+from shapely import wkb, wkt
+import typer
+
+import dcpy.models.product.dataset.metadata as models
+from dcpy.utils.logging import logger
+
+
+class ErrorType(Enum):
+    INVALID_DATA = "INVALID_DATA"
+    NULLS_FOUND = "NULLS_FOUND"
+    COLUMM_MISMATCH = "COLUMN_MISMATCH"
+    INVALID_METADATA = "INVALID_METADATA"
+    UNHANDLED_EXCEPTION = "UNHANDLED_EXCEPTION"
+    MISSING_FILE = "MISSING_FILE"
+
+
+@dataclass
+class ValidationError:
+    error_type: ErrorType
+    message: str
+    dataset_file: models.DatasetFile | None
+
+
+@dataclass
+class ValidationStats:
+    row_count: int | None
+
+
+@dataclass
+class DatasetFileValidation:
+    stats: ValidationStats = field(
+        default_factory=lambda: ValidationStats(row_count=None)
+    )
+    errors: list[ValidationError] = field(default_factory=lambda: [])
+
+
+@dataclass
+class PackageValidation:
+    validations: list[DatasetFileValidation]
+    errors: list[ValidationError]
+
+    def get_dataset_errors(self) -> list[ValidationError]:
+        """Get validation errors from all dataset validations in the package."""
+        return sum([v.errors for v in self.validations if v.errors], [])
+
+    def pretty_print_errors(self):
+        ds_errors = self.get_dataset_errors()
+
+        for e in ds_errors + self.errors:
+            pprint.pp(
+                [
+                    e.dataset_file.filename if e.dataset_file else None,
+                    e.error_type.value,
+                    e.message,
+                ]
+            )
+
+
+def _is_valid_wkb(g):
+    if not g:
+        return True
```

Comment:
seems like this should `return False` right?

Reply:
So in validation, I'd separated out the null checks from the type-checks. This is really just meant to remove non-null invalids, so they're not fed into wkb.loads